### PR TITLE
[BEAM-1406] Removes deprecated fileio.TextFileSink

### PR DESCRIPTION
**Pull request by chamikaramj**
Users should be using textio.WriteToText() transform instead of
fileio.TextFileSink.

### CODE REVIEW

**Comment by chamikaramj** [comment]
R: @aaltay

File: sdks/python/apache_beam/__init__.py [hunk 1]
```diff
@@ -53,8 +53,8 @@
 
 * transform classes are available as beam.FlatMap, beam.GroupByKey, etc.
 * Pipeline class is available as beam.Pipeline
-* text source/sink classes are available as beam.io.TextFileSource,
-  beam.io.TextFileSink
+* text source/sink transforms are available as beam.io.RedfromText,
```
Comment (sb2nov) [inline sdks/python/apache_beam/__init__.py]: typo in Read

File: sdks/python/apache_beam/io/textio.py [hunk 2]
```diff
@@ -214,9 +214,73 @@ def _read_record(self, file_to_read, read_buffer):
...
+
+  def __init__(self,
+               file_path_prefix,
+               file_name_suffix='',
+               append_trailing_newlines=True,
+               num_shards=0,
+               shard_name_template=None,
+               coder=coders.ToStringCoder(),
+               compression_type=fileio.CompressionTypes.AUTO):
+    """Initialize a TextFileSink.
```
Comment (aaltay) [inline sdks/python/apache_beam/io/textio.py]: Comment update to: `_TextSink`

File: sdks/python/apache_beam/io/textio.py [hunk 3]
```diff
@@ -214,9 +214,73 @@ def _read_record(self, file_to_read, read_buffer):
...
+        generated. The default pattern used is '-SSSSS-of-NNNNN'.
+      coder: Coder used to encode each line.
+      compression_type: Used to handle compressed output files. Typical value
+          is CompressionTypes.AUTO, in which case the final file path's
+          extension (as determined by file_path_prefix, file_name_suffix,
+          num_shards and shard_name_template) will be used to detect the
+          compression.
+
+    Returns:
+      A TextFileSink object usable for writing.
```
Comment (aaltay) [inline sdks/python/apache_beam/io/textio.py]: Commet update: `_TextSink`

File: sdks/python/apache_beam/__init__.py (hunk 1 above)
Comment (aaltay) [inline sdks/python/apache_beam/__init__.py]: `read/write transforms` instead of `source/sink transforms`?

Comment (chamikaramj) [inline sdks/python/apache_beam/__init__.py]: Done.

Comment (chamikaramj) [inline sdks/python/apache_beam/__init__.py]: Done.

File: sdks/python/apache_beam/io/textio.py (hunk 2 above)
Comment (chamikaramj) [inline sdks/python/apache_beam/io/textio.py]: Done.

File: sdks/python/apache_beam/io/textio.py (hunk 3 above)
Comment (chamikaramj) [inline sdks/python/apache_beam/io/textio.py]: Done.

**Comment by chamikaramj** [comment]
Thanks. PTAL.

**Comment by aaltay** [comment]
LGTM

**Pull request closed without merging**